### DISCUSSION TITLE
not connecting socket.io nextjs app with nodejs app

### DISCUSSION THREAD

**Discussion by Permanf** [socketio/socket.io#4933] [Q&A] [answered]
Hello everybody,
I am trying to integrate Socket.io with (Nextjs app, Nodejs app) and I'm having difficulties making a connection from the client-side to the server. I've looked through other related questions but my issue is happening locally, so there's no web server in the middle.
WebSocket connection to 'ws://localhost:8000/socket.io/?EIO=4&transport=websocket' failed:
how to fixed that?
Client-side
```js
// Create a socket connection
    const socket = io("http://localhost:8000", {transports: ["websocket"]});
    socket.on("connection", () => {
        console.log(socket.id);
    });
    socket.on("disconnect", () => {
        console.log(socket.id); // undefined
    });
```
Server
```js
const httpServer = require("http").createServer();
const io = require("socket.io")(httpServer, {
  cors: {
    origin: "http://localhost:4010",
  }
});

// function that handles socket.io connections
io.on('connection', (socket) => {
  // on connect print user connected msg
  console.log('A user connected.');

  // on socket disconnect
  socket.on('disconnect', () => {
      // print user disconnect msg
      console.log('A user disconnected.');
  });

  // Handle custom 'hello' event
  socket.on('message', (data) => {
      console.log('Server Received hello event:', data);
      // Emit a response event
      socket.emit('response', 'Hello, client!');
  });
});
```

Edit by @darrachequesne: code formatting

**Comment by darrachequesne** [comment, marked as answer]
That's weird! If you go to http://localhost:8000/socket.io/?EIO=4&transport=polling, does it reach the Socket.IO server?

Reference: https://socket.io/docs/v4/troubleshooting-connection-issues/#troubleshooting-steps
